unit-test-security-rules-v9/test/database.spec.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const { readFileSync, createWriteStream } = require('fs');
+const http = require('http');
+
+const testing = require('@firebase/rules-unit-testing');
+const { initializeTestEnvironment, assertFails, assertSucceeds } = testing;
+
+const { ref, get, update } = require('firebase/database');
+
+/** @type testing.RulesTestEnvironment */
+let testEnv;
+/** @type testing.RulesTestContext */
+let unauthedDb;
+/** @type testing.RulesTestContext */
+let aliceDb;
+/** @type testing.RulesTestContext */
+let bobDb;
+
+before(async () => {
+  testEnv = await initializeTestEnvironment({
+    database: {rules: readFileSync('database.rules.json', 'utf8')},
+  });
+
+  unauthedDb = testEnv.unauthenticatedContext().database();
```

Comment:
nit: I'd suggest inlining these to make each snippet more standalone and copypasteable. ([Tests too DRY? Make them DAMP!](https://testing.googleblog.com/2019/12/testing-on-toilet-tests-too-dry-make.html)) Creating an instance per test also reduces chances of cache-related issues.